core/common/src/main/java/alluxio/client/file/URIStatus.java

```diff
@@ -310,17 +313,18 @@ public String getUfsFingerprint() {
   }
 
   /**
-   * @return the cache quota
+   * @return the Presto context
    */
-  public CacheQuota getCacheQuota() {
-    return mInfo.getCacheQuota();
+  @Nullable
+  public PrestoContext getPrestoContext() {
+    return mPrestoContext;
   }
 
   /**
-   * @return the cache scope
+   * @param prestoContext Presto context to set
    */
-  public CacheScope getCacheScope() {
-    return mInfo.getCacheScope();
+  public void setPrestoContext(PrestoContext prestoContext) {
```

Comment:
from where will this be called?

Reply:
this will be only called in Presto land, likely https://github.com/prestodb/presto/blob/master/presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingFileSystem.java#L80

Reply:
I have the similar question.   Why do we need a setter for PrestoContext? 

The existing calling chain is starting from AlluxioCachingFileSystem::openFile
```
if (hiveFileContext.getCacheQuota().getQuota().isPresent()) {
                    info.setCacheQuota(new CacheQuota(ImmutableMap.of(scope.level(), hiveFileContext.getCacheQuota().getQuota().get().toBytes())));
                }
                else {
                    info.setCacheQuota(CacheQuota.UNLIMITED);
                }
``` 
Can we put PrestoContext as a a parameter in the constructor of this class just like FileInfo?